src/modules/deal-investor/deal-investor.controller.test.ts

```diff
@@ -61,4 +70,36 @@ describe('DealInvestorController', () => {
       expect(dealInvestors).toStrictEqual(expectedDealInvestorsWithNewKey);
     });
   });
+
+  describe('createInvestorForDeal', () => {
+    const dealIdentifier: UkefId = valueGenerator.ukefId();
+    const portfolioIdentifier = valueGenerator.string();
+
+    const { requestBodyToCreateDealInvestor } = new CreateDealInvestorGenerator(valueGenerator, dateStringTransformations).generate({
+      numberToGenerate: 2,
+      dealIdentifier: dealIdentifier,
+      portfolioIdentifier: portfolioIdentifier,
+    });
+
+    it('creates an investor for the deal with the service from the request body', async () => {
+      await controller.createInvestorForDeal(dealIdentifier, requestBodyToCreateDealInvestor);
+
+      expect(dealInvestorServiceCreateInvestorForDeal).toHaveBeenCalledWith(dealIdentifier, requestBodyToCreateDealInvestor[0]);
+    });
+
+    it('returns the deal identifier if creating the investor succeeds', async () => {
+      const response = await controller.createInvestorForDeal(dealIdentifier, requestBodyToCreateDealInvestor);
+
+      expect(response).toStrictEqual(new CreateDealInvestorResponse(dealIdentifier));
+    });
+
+    it('does NOT include unexpected keys from the request body', async () => {
+      const requestBodyToCreateDealInvestorPlusUnexpectedKeys = JSON.parse(JSON.stringify(requestBodyToCreateDealInvestor));
```

Comment:
My thinking was to create a deep copy of `requestBodyToCreateDealInvestor` so we are not modifying the that object itself when we add the unexpected key

Reply:
Oh I see, good thinking on needing a copy.

Personally I prefer
```javascript
const requestBodyToCreateDealInvestorPlusUnexpectedKeys = [{
  ...requestBodyToCreateDealInvestor[0],
  unexpectedKey: 'unexpected value',
}];
```
or 
```javascript
const requestBodyToCreateDealInvestorPlusUnexpectedKeys = [{
  ...requestBodyToCreateDealInvestor[0],
  unexpectedKey: 'unexpected value',
}, requestBodyToCreateDealInvestor[1]];
```
because it creates a new object instead of modifying an existing one

Reply:
Good shout, I've gone with the latter of your suggestions.

Reply:
@abhi-markan in Luke's absence would you be able to resolve this thread if you are happy please, just so I can get this merged in today